### ISSUE TITLE
moe model export error

### ISSUE THREAD

**Issue by yspMing** [NVIDIA-NeMo/Megatron-Bridge#2232]
**Describe the bug**

Not sure about the moe model export (torch_dist -> hf) is right. 

Suppose we have a basic hf model(moe) hf1, and we convert it to torch_dist1 for pretrain/sft;

To ensure to the model export is right, we convert torch_dist1 back to hf model (hf2), and suppose hf1 and hf2 is the same, however using hf2 to train will lead to different training loss and step time (moe load balance)

**Steps/Code to reproduce bug**

model：
qwen3-235b-a22b-l4 235B models with only 4 layers

# convert to torch_dist and train several steps
python examples/conversion/convert_checkpoints.py import \
  --hf-model /Path/to/Qwen3-235B-A22B-4-layers/ \
  --megatron-path /Path/to/Qwen3-235B-A22B-4-layers-dist \
  --torch-dtype bfloat16 \
  --device-map auto

torchrun finetune.py --pretrained-checkpoint /Path/to/Qwen3-235B-A22B-4-layers-dist 

########################
# convert to hf2 and train several steps
python examples/conversion/convert_checkpoints.py export \
  --hf-model /Path/to/Qwen3-235B-A22B-4-layers/ \
  --megatron-path /Path/to/Qwen3-235B-A22B-4-layers-dist \
  --hf-path /Path/to/temporary_hf_model

python examples/conversion/convert_checkpoints.py import \
  --hf-model /Path/to/temporary_hf_model \
  --megatron-path /Path/to/temporary_torchdist_model \
  --torch-dtype bfloat16 \
  --device-map auto

torchrun finetune.py --pretrained-checkpoint /Path/to/temporary_torchdist_model 

The above two trainings have gaps

**Comment by yspMing** [comment]
Seems I have different training configs in 2nd training, never mind....

**Comment by pruprakash** [comment]
## QA RCCA Analysis

**Classification**: USER ERROR

**Reason**: User confirmed: 'Seems I have different training configs in 2nd training, never mind...'

**RCCA Required**: NO - This is not a code bug requiring root cause corrective action.
